lib/pairmotron/pair_maker.ex

```diff
@@ -47,8 +47,12 @@ defmodule Pairmotron.PairMaker do
 
     determination = PairBuilder.determify(pairs, users)
 
-    determination.dead_pairs
-      |> Enum.map(fn(p) -> Repo.delete! p end)
+    multi = Ecto.Multi.new
+
+    multi = determination.dead_pairs
+      |> Enum.reduce(multi, fn(pair, acc) ->
+          acc |> Ecto.Multi.delete(:remove_dead_pairs, pair)
```

Comment:
See an alternative way of doing this? Since we have to apply "actions" on a multi, it acts exactly like an accumulator.